src/test/shared/utilities/vsCodeUtils.test.ts

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+
+import { BasicLogger, ErrorOrString, LogLevel } from '../../../shared/logger'
+import { getChannelLogger, localize, processTemplate, TemplateParams } from '../../../shared/utilities/vsCodeUtils'
+
+import { MockOutputChannel } from '../../mockOutputChannel'
+
+class MockLogger implements BasicLogger {
+
+    public logs = {
+        verbose: new Set<string>(),
+        debug: new Set<string>(),
+        info: new Set<string>(),
+        warn: new Set<string>(),
+        error: new Set<string>(),
+    }
+    public outputChannel?: vscode.OutputChannel
+    public verbose(...messages: ErrorOrString[]) {
+        this.logs.verbose.add(MockLogger.format(messages))
+    }
+    public debug(...messages: ErrorOrString[]) {
+        this.logs.debug.add(MockLogger.format(messages))
+    }
+    public info(...messages: ErrorOrString[]) {
+        this.logs.info.add(MockLogger.format(messages))
+    }
+    public warn(...messages: ErrorOrString[]) {
+        this.logs.warn.add(MockLogger.format(messages))
+    }
+    public error(...messages: ErrorOrString[]) {
+        this.logs.error.add(MockLogger.format(messages))
+    }
+    public static format(messages: ErrorOrString[]) {
+        return JSON.stringify(messages.map(msg => msg instanceof Error ? msg.message : msg))
+    }
+}
+
+describe('vsCodeUtils getChannelLogger', async () => {
+
+    let logger: MockLogger
+    let outputChannel: MockOutputChannel
+
+    beforeEach(async () => {
+        logger = new MockLogger()
+        outputChannel = new MockOutputChannel()
+    })
+
+    const testWith = ({
+        title, nlsKey, nlsTemplate, templateTokens = []
+    }: TemplateParams & { title: string }) => {
+        describe(title, function() {
+            const logLevels: LogLevel[] = ['verbose', 'debug', 'info', 'warn', 'error']
+            logLevels.forEach( level => {
+                it(`with ${level} `, function() {
```

Comment:
This test appears to verify several things. Ideally each test verifies one thing. Can this be broken up into simpler tests that each verify a single piece of functionality?

Reply:
✔️